corehq/apps/userreports/reports/filters/choice_providers.py

```diff
@@ -402,28 +403,33 @@ def default_value(self, user):
 class DomainChoiceProvider(ChainableChoiceProvider):
 
     @memoized
-    def _query_domains(self, domain, query_text):
+    def _query_domains(self, domain, query_text, user):
         domains = {domain}
+        if user is None:
+            return list(domains)
+        if not RegistryPermissionCheck(domain, user).can_view_registry_data(
+                self.report.registry_helper.registry_slug):
+            return list(domains)
```

Comment:
nit: you can combine the 2 if statements
```suggestion
        if not (user and RegistryPermissionCheck(domain, user).can_view_registry_data(
                self.report.registry_helper.registry_slug)):
            return list(domains)
```